### PR TITLE
CIRC-1450 Update logic for checkout request update

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
- Update logic for checkout request update;
- Create test;

Resolves:  [CIRC-1450](https://issues.folio.org/browse/CIRC-1450)
## Approach
- Ignore request update if the request is not associated with the item

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java [hunk 1]
```diff
@@ -171,13 +171,18 @@ public static UpdateRequestQueue using(Clients clients,
...
       .thenApply(result -> result.map(relatedRecords::withRequestQueue));
   }
 
-  private CompletableFuture<Result<RequestQueue>> onCheckOut(RequestQueue requestQueue) {
+  private CompletableFuture<Result<RequestQueue>> onCheckOut(RequestQueue requestQueue,
+    String itemId) {
+
     if (requestQueue.hasOutstandingFulfillableRequests()) {
       Request firstRequest = requestQueue.getHighestPriorityFulfillableRequest();
+      if (!firstRequest.getItemId().equals(itemId)) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java:183]: firstRequest.getItemId() - can this be null?

Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java]: updated

File: src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java [hunk 2]
```diff
@@ -171,13 +171,18 @@ public static UpdateRequestQueue using(Clients clients,
   public CompletableFuture<Result<LoanAndRelatedRecords>> onCheckOut(
     LoanAndRelatedRecords relatedRecords) {
 
-    return onCheckOut(relatedRecords.getRequestQueue())
+    return onCheckOut(relatedRecords.getRequestQueue(), relatedRecords.getItem().getItemId())
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java:174]: Can relatedRecords.getItem() be null? Or relatedRecords.getItem().getItemId()?

File: src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java [hunk 3]
```diff
@@ -171,13 +171,18 @@ public static UpdateRequestQueue using(Clients clients,
...
       .thenApply(result -> result.map(relatedRecords::withRequestQueue));
   }
 
-  private CompletableFuture<Result<RequestQueue>> onCheckOut(RequestQueue requestQueue) {
+  private CompletableFuture<Result<RequestQueue>> onCheckOut(RequestQueue requestQueue,
+    String itemId) {
+
     if (requestQueue.hasOutstandingFulfillableRequests()) {
       Request firstRequest = requestQueue.getHighestPriorityFulfillableRequest();
+      if (!itemId.equals(firstRequest.getItemId())) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java:183]: itemId can be null - someone may pass null to this method. For example, if relatedRecords.getItem().getItemId() is null on line 174

**Pull request merged**